sys/mips/include/param.h

```diff
@@ -207,8 +225,8 @@
 /*
  * Mach derived conversion macros
  */
-#define	round_2mpage(x)		(((x) + PDRMASK) & ~PDRMASK)
-#define	trunc_2mpage(x)		((x) & ~PDRMASK)
+#define	round_2mpage(x)		__builtin_align_down(((x) + PDRMASK), PDRSIZE)
+#define	trunc_2mpage(x)		__builtin_align_down((x), PDRSIZE)
```

Comment:
```suggestion
#define	round_2mpage(x)		roundup2((x), PDRSIZE)
#define	trunc_2mpage(x)		rounddown2((x), PDRSIZE)
```
which is upstreamable.